.github/workflows/ci.yml

```diff
@@ -0,0 +1,125 @@
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.3', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.4', node: '14.15', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_39_STABLE', php: '7.4', node: '14.15', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_310_STABLE', php: '7.4', node: '14.15.1', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_310_STABLE', php: '7.4', node: '14.15.1', database: 'pgsql'}
+          - {moodle-branch: 'MOODLE_311_STABLE', php: '7.4', node: '14.15.1', database: 'mariadb'}
+          - {moodle-branch: 'MOODLE_311_STABLE', php: '7.4', node: '14.15.1', database: 'pgsql'}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          # Clone in plugin subdir, so we can setup CI in default directory.
+          path: plugin
+
+      - name: Install node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Setup PHP environment
+        uses: shivammathur/setup-php@v2 #https://github.com/shivammathur/setup-php
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: mbstring, pgsql, mysqli, gd, zip
+          tools: phpunit
+
+      - name: Deploy moodle-plugin-ci
+        run: |
+          composer create-project -n --no-dev --prefer-dist moodlehq/moodle-plugin-ci ci ^3
+          # Use composer V1 for older branches
+          composer self-update --1
+          # Add dirs to $PATH
+          echo $(cd ci/bin; pwd) >> $GITHUB_PATH
+          echo $(cd ci/vendor/bin; pwd) >> $GITHUB_PATH
+          # PHPUnit depends on en_AU.UTF-8 locale
+          sudo locale-gen en_AU.UTF-8
+
+      - name: Install Moodle
+        # Need explicit IP to stop mysql client fail on attempt to use unix socket.
+        run: moodle-plugin-ci install -vvv --plugin ./plugin --db-host=127.0.0.1
+        env:
+          DB: ${{ matrix.database }}
+          # TODO: Omitted MOODLE_BRANCH results in regex failure, investigate.
+          MOODLE_BRANCH: ${{ matrix.moodle-branch }}
+
+      - name: Run phplint
+        if: ${{ always() }}
+        run: moodle-plugin-ci phplint
+
+      - name: Run phpcpd
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpcpd || true
+
+      - name: Run phpmd
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpmd
+
+      - name: Run codechecker
+        if: ${{ always() }}
+        run: moodle-plugin-ci codechecker
+
+      - name: Run validate
+        if: ${{ always() }}
+        run: moodle-plugin-ci validate
+
+      - name: Run savepoints
+        if: ${{ always() }}
+        run: moodle-plugin-ci savepoints
+
+      # PHPDoc Needs a lot of documenting work
+      #- name: Run phpdoc
+      #  if: ${{ always() }}
+      #  run: moodle-plugin-ci phpdoc
+
+      # It was initially done for shifter, needs adaptation for grunt
```

Comment:
Could you please create an issue for this (if it doesn't exists) and add some details